packages/leancode_hooks/lib/src/use_stream_listener.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+
+/// Listens to [stream] and calls [onData] on new data.
+///
+/// See also:
+///
+/// * [Stream.listen] for information about [onError], [onDone], and
+///   [cancelOnError].
+void useStreamListener<T>(
+  Stream<T> stream,
+  ValueChanged<T> onData, {
+  Function? onError,
+  void Function()? onDone,
+  bool? cancelOnError,
+  List<Object?> keys = const [],
+}) {
+  useEffect(
+    () {
+      final subscription = stream.listen(
+        onData,
+        onError: onError,
+        onDone: onDone,
+        cancelOnError: cancelOnError,
+      );
+
+      return subscription.cancel;
+    },
+    keys,
+  );
+}
```

Comment:
I improved it in the way we talked about on weekly meeting on Friday. 

I thought that, given that `package:flutter_hooks` already has [useStream](https://pub.dev/documentation/flutter_hooks/latest/flutter_hooks/useStream.html) and [useStreamController](https://pub.dev/documentation/flutter_hooks/latest/flutter_hooks/useStreamController.html), we could contribute our `useStreamListener` to it.

[Looks like nobody has thought about it yet before :P](https://github.com/rrousselGit/flutter_hooks/issues?q=usestreamlistener)